docs/custom/error-handlers.mdx

```diff
@@ -0,0 +1,154 @@
+---
+id: error_handlers
+title: Custom error handlers
+---
+
+:::caution
+This AMF functionality is currently a work in progress. The following features are being developed:
+- Support for custom error handling in parsing, validation and emission stages
+- Javascript support
+:::
```

Comment:
An error handler is a mechanism that lets you handle any errors that arise in AMF. You can define custom error handlers and use them as plugins in AMF's resolution stage.

## AMF error types

There are 3 types of errors in AMF, denoted by the `amf.core.validation.SeverityLevels` object:

- **Info**
  - Indicates something may or may not be wrong
  - Typically informational depending on user intent 
- **Warning**
  - Indicates that something is not correct and action is required
  - Includes warnings about future changes, tool deprecations, or errors that will become violations in the future
- **Violation**
  - Indicates that something is in violation of normal operations or is not compliant with its specification
  - Invalidates the AMF graph model